src/Aspire.Hosting.Azure/Provisioning/Provisioners/BicepProvisioner.cs

```diff
@@ -308,6 +307,21 @@ await notificationService.PublishUpdateAsync(resource, state =>
         .ConfigureAwait(false);
     }
 
+    private void ConfigureSecretResolver(IAzureKeyVaultResource kvr)
+    {
+        var resource = (AzureBicepResource)kvr;
+
+        var vaultUri = resource.Outputs[kvr.VaultUriOutputReference.Name] as string ?? throw new InvalidOperationException($"{kvr.VaultUriOutputReference.Name} not found in outputs.");
+
+        // Set the client for resolving secrets at runtime
+        var client = new SecretClient(new(vaultUri), tokenCredentialHolder.Credential);
```

Comment:
For my knowledge, this didn't really need to change to take a `tokenCredentialHolder`, did it? You could get the same credential from the `context.Credential` as before, right?

Reply:
It did, this method didn't take the context.